src/stories/TextInput.tsx

```diff
@@ -0,0 +1,161 @@
+//on render, component needs the cursor position
+
+import React, { useRef, useState, useEffect } from "react";
+import InfoIcon from "@mui/icons-material/Info";
+import "./textinput.css";
+import TitleIcon from "@mui/icons-material/Title";
+import KeyboardReturnIcon from "@mui/icons-material/KeyboardReturn";
+
+interface TextInputProps {
+  placeholder: string;
+  isDisabled: boolean;
+  isPassword: boolean;
+  validationRegex: RegExp | string;
+}
+
+const TextInput = ({
+  placeholder,
```

Comment:
this should be a controllable textInput - 

it will need to take a value and setValue